### ISSUE TITLE
Multimonitor issue

### ISSUE THREAD

**Issue by DonHugo69** [ajoshpratt/kwinOverview#1]
Hey there, 

great work you have done with your script!
However, I am encountering an issue in my multimonitor setup, shown below:
![grafik](https://user-images.githubusercontent.com/10484857/40267172-6ea7e55c-5b58-11e8-8afa-4784e546c869.png)

What happens is, that the overview is not only drawn on the left, primary monitor but takes the increased height of the right one as a starting point:
![grafik](https://user-images.githubusercontent.com/10484857/40267184-c8769f60-5b58-11e8-8465-b7e8a4d8e3fd.png)

Thus, the upper part is missing and not usable.

**Comment by ajoshpratt** [comment]
That's a fairly amusing looking screenshot, although I imagine quite frustrating : )

I no longer have a second monitor to test, so I'm surprised the code in there even functions with a second monitor as it is.  I'm not surprised that it only shows up on the primary monitor, but it looks like I need to at least tweak the calculation for determining where to place the dashboard.

It looks like the addition of a second monitor positions 0,0 at a different point in the screen space.  I'll try to implement some fixes, if you wouldn't mind testing them once they're available.

**Comment by DonHugo69** [comment]
Thanks a lot for your quick response. 
I think you are right that 0,0 is not aligned with the upper left of the primary monitor. Looking at a fullscreen screenshot
![grafik](https://user-images.githubusercontent.com/10484857/40270983-32f57364-5b97-11e8-999e-d0591e083b7b.png)
one can see the black space. I don't know the underlying mechanisms but I guess it is easier to always create a rectangular drawing surface for the windows. 

Anyways, I'll happily test the fixes as soon as they are implemented. This is probably the last extension to make my KDE experience nearly perfect.

**Comment by ajoshpratt** [comment]
@DonHugo69, can you try out the multiscreen branch?  Although it doesn't respond on the fly to changes in geometry/positioning, it should at least respect however the screens are set up when KWin is launched.

https://github.com/ajoshpratt/kwinOverview/tree/multiscreen

If you change orientation, you'll currently have to restart KWin, which isn't a great long term solution, but does at least work.

I tested this mostly by hooking up both outputs from my video card to my monitor, so lemme know if this works in a real world scenario!

**Comment by FrauH0lle** [comment]
It looks better but not perfect:
![grafik](https://user-images.githubusercontent.com/10484857/40283272-f8c2d28c-5c7b-11e8-9116-5c5613754de8.png)
The script takes the wrong monitor.

If there are any debug I can provide, just tell me where to look or which parameters to set up.

**Comment by astatide** [comment]
Out of curiosity, which monitor was active when you enabled the overview?

It looks like I'll probably need to attach to the 'screen changed' signal and adjust screen geometry and position.  I made some attempts to do so when I had time last weekend, but couldn't properly debug it at the time.  I'll try sorting that out soon.

**Comment by FrauH0lle** [comment]
Indeed, I noticed that if I start the right monitor after the left one, the overview appears on the left (primary) one.

I can observe a similar behavior for Latte Dock, which initializes on the right screen and then moves itself to the primary one on startup.
